src/bin/train.cc

```diff
@@ -456,49 +480,74 @@ int main(int argc, char * argv[]) {
         trainer->update(lr_decay);
 
         num_trained_samples += batch.source_ids[0].size();
+        num_trained_words += batch.target_ids.size() * batch.target_ids[0].size();
```

Comment:
Some comments are required because this calculation also includes the number of padding words.